tests/unit/utils/test_event_matcher.py

```diff
@@ -0,0 +1,124 @@
+import json
+
+import pytest
+
+from localstack import config
+from localstack.utils.event_matcher import matches_event
+
+EVENT_PATTERN_DICT = {
+    "source": ["aws.ec2"],
+    "detail-type": ["EC2 Instance State-change Notification"],
+}
+EVENT_DICT = {
+    "source": "aws.ec2",
+    "detail-type": "EC2 Instance State-change Notification",
+    "detail": {"state": "running"},
+}
+EVENT_PATTERN_STR = json.dumps(EVENT_PATTERN_DICT)
+EVENT_STR = json.dumps(EVENT_DICT)
+
+
+@pytest.fixture
+def event_rule_engine(monkeypatch):
+    """Fixture to control EVENT_RULE_ENGINE config"""
+
+    def _set_engine(engine: str):
+        monkeypatch.setattr(config, "EVENT_RULE_ENGINE", engine)
+
+    return _set_engine
+
+
+def test_matches_event_with_java_engine_strings(event_rule_engine):
+    """Test Java engine with string inputs (EventBridge case)"""
+    event_rule_engine("java")
```

Comment:
@dominikschubert @dfangl Should we avoid any JPype-starting monkey-patched tests in the LS CI?
I'm somewhat worried this will blow up once we combine it with other usage (e.g., SFN tests)?

This shouldn't block the merge here. I'm happy to tackle this in a follow-up because we have several other usages.